### PR TITLE
Feature/kiwi image description2

### PR DESCRIPTION
**Pull request by evanrolfe**
https://trello.com/c/6c513ZjA/605-3-p1-edit-sections-image-and-description

Allows you to edit:
* image name
* author
* contact
* specification

Screenshots:
screenshot from 2017-10-27 09-00-46

screenshot from 2017-10-27 09-01-00

### CODE REVIEW

File: src/api/app/models/kiwi/image/xml_builder.rb [hunk 1]
```diff
@@ -13,11 +13,29 @@ def build
 
         doc = update_packages(doc)
         doc = update_repositories(doc)
+        doc = update_description(doc) if @image.description.present?
```
Comment (mdeniz) [inline src/api/app/models/kiwi/image/xml_builder.rb]: I would move the `if` check inside the method as a guard clause. The method should know what to do by itself.

```ruby
def update_description(document)
  document unless @image.description.present?
  #...

end

File: src/api/app/controllers/webui/kiwi/images_controller.rb [hunk 2]
```diff
@@ -31,7 +31,9 @@ def import_from_package
       end
 
       def show
+        @image.build_description if @image.description.nil? # Because the form needs a Description object instantiated
```
Comment (mdeniz) [inline src/api/app/controllers/webui/kiwi/images_controller.rb]: Isn't better... and is more self explanatory, that way maybe you can remove the comment
```ruby
@image.build_description unless @image.description.present?
```

Comment (evanrolfe) [reply src/api/app/controllers/webui/kiwi/images_controller.rb]: I'm pretty sure that rubocop will tell you to use `if` over `unless` here. Also the comment explains why this is necessary, and changing the code like you suggested still doesn't explain that.

File: src/api/app/models/kiwi/image/xml_builder.rb (hunk 1 above)
Comment (evanrolfe) [reply src/api/app/models/kiwi/image/xml_builder.rb]: Done.

**Comment by hennevogel** [comment]
I guess we have the first winner of a codeclimate slap. Mr. Rolfe! 🏆

**Comment by evanrolfe** [comment]
@mdeniz Please re-review

File: src/api/app/models/kiwi/image/xml_builder.rb [hunk 3]
```diff
@@ -13,11 +13,37 @@ def build
...
       end
 
       private
 
+      def update_description(document)
+        return document if @image.description.blank?
+
+        if document.xpath('image/description[@type="system"]').first
+          %w[author contact specification].each do |element_name|
+            update_description_element(document, element_name)
```
Comment (mdeniz) [inline src/api/app/models/kiwi/image/xml_builder.rb]: Side effects in the document!?? I don't like that... see the other comment.

File: src/api/app/models/kiwi/image/xml_builder.rb [hunk 4]
```diff
@@ -13,11 +13,37 @@ def build
...
+        document
+      end
+
+      def update_description_element(document, element_name)
+        element_xml = document.xpath("image/description[@type='system']/#{element_name}").first
+        unless element_xml
+          element_xml = Nokogiri::XML::Node.new(element_name, document)
+          document.xpath('image/description[@type="system"]').first.add_child(element_xml)
+        end
+        # rubocop:disable Lint/UselessSetterCall
```
Comment (mdeniz) [inline src/api/app/models/kiwi/image/xml_builder.rb]: Why don't you do this in 2 steps, first calculate the element_xml in one method and then update the content outside..

Comment (evanrolfe) [reply src/api/app/models/kiwi/image/xml_builder.rb]: Good idea, fixed.

**Pull request merged**